metadata-ingestion/src/datahub/sql_parsing/sqlglot_lineage.py

```diff
@@ -158,6 +161,9 @@ class ColumnLineageInfo(_ParserBaseModel):
 
     logic: Optional[ColumnTransformation] = pydantic.Field(default=None)
 
+    def __hash__(self) -> int:
+        return hash((self.downstream, tuple(self.upstreams), self.logic))
```

Comment:
I'm ok allowing these hash functions provided that we include a large TODO comment explaining (1) that this is not a good practice and we want to remove it and (2) why we needed to add them in the first place and (3) any updates to the class must also update the hash function

Reply:
Done